### PR TITLE
Feature/pn 310 ttl change to timestamp

### PR DESCRIPTION
**Pull request by bogh**

### CODE REVIEW

File: protocol/message_test.go [hunk 1]
```diff
@@ -223,7 +220,14 @@ func TestMessage_IsExpired(t *testing.T) {
 	}
 
 	for i, c := range cases {
-		a.Equal(c.result, (&Message{Expires: &c.expires}).IsExpired(), "Failed IsExpired case: %d", i)
+		a.Equal(c.result, (&Message{Expires: c.expires.Unix()}).IsExpired(), "Failed IsExpired case: %d", i)
 	}
 
 }
+
+func TestMessage_IsExpired_WithNilExpires(t *testing.T) {
```
Comment (cosminrentea) [inline protocol/message_test.go]: WithoutExplicitExpires

File: protocol/message_test.go [hunk 2]
```diff
@@ -10,15 +10,15 @@ import (
 )
 
 var (
-	aNormalMessage = `/foo/bar,42,user01,phone01,{"user":"user01"},2017-03-17T20:04:26+02:00,1420110000,1
+	aNormalMessage = `/foo/bar,42,user01,phone01,{"user":"user01"},1420110000,1420110000,1
 {"Content-Type": "text/plain", "Correlation-Id": "7sdks723ksgqn"}
 Hello World`
 
-	aNormalMessageNoExpires = `/foo/bar,42,user01,phone01,{"user":"user01"},,1420110000,1
+	aNormalMessageNoExpires = `/foo/bar,42,user01,phone01,{"user":"user01"},0,1420110000,1
```
Comment (cosminrentea) [inline protocol/message_test.go]: should be ,, instead of ,0,    ?

File: protocol/message.go [hunk 3]
```diff
@@ -146,13 +144,11 @@ func (m *Message) SetFilter(key, value string) {
...
-//
-// Checks are made using `Expires` field timezone
 func (m *Message) IsExpired() bool {
-	if m.Expires == nil {
-		return true
+	if m.Expires == 0 {
+		return false
 	}
-	return m.Expires != nil && m.Expires.Before(time.Now().In(m.Expires.Location()))
+	return m.Expires != 0 && m.Expires < time.Now().Unix()
```
Comment (cosminrentea) [inline protocol/message.go]: we already know at this point in code that m.Expires != 0
condition can be simplified

**Review by cosminrentea — CHANGES_REQUESTED** [review]

File: protocol/message.go (hunk 3 above)
Comment (bogh) [reply protocol/message.go]: Sorry. This was there before adding the condition above.

File: protocol/message_test.go [hunk 4]
```diff
@@ -267,7 +264,14 @@ func TestMessage_IsExpired(t *testing.T) {
...
-		a.Equal(c.result, (&Message{Expires: &c.expires}).IsExpired(), "Failed IsExpired case: %d", i)
+		a.Equal(c.result, (&Message{Expires: c.expires.Unix()}).IsExpired(), "Failed IsExpired case: %d", i)
 	}
 
 }
+
+func TestMessage_IsExpired_WithoutExplicitExpires(t *testing.T) {
+	a := assert.New(t)
+
+	a.Equal(false, (&Message{}).IsExpired())
```
Comment (marian-craciunescu) [inline protocol/message_test.go]: you can use directly a.False((&Message{}).IsExpired()).

File: server/sms/nexmo_sms_sender.go [hunk 5]
```diff
@@ -138,6 +138,17 @@ func (ns *NexmoSender) Send(msg *protocol.Message) error {
 		return ErrRetryFailed
 	}
 
+	if msg.IsExpired() {
+		log.WithFields(log.Fields{
+			"ID":      msg.ID,
+			"To":      nexmoSMS.To,
+			"Expires": msg.Expires,
```
Comment (marian-craciunescu) [inline server/sms/nexmo_sms_sender.go]: maybe also put msg.Expires in a human readable format like RFC3339

**Review by marian-craciunescu — CHANGES_REQUESTED** [review]

**Review by marian-craciunescu — APPROVED** [review]

**Pull request closed without merging**